dd-sdk-android-ndk/src/main/cpp/utils/signal-monitor.c

```diff
@@ -194,28 +192,25 @@ bool init_datadog_signal_handler() {
     if (datadog_sigaction == NULL) {
         return false;
     }
-    sigemptyset((sigset_t *) &datadog_sigaction->sa_mask);
+    if (sigemptyset((sigset_t *) &datadog_sigaction->sa_mask) != 0) {
+        return false;
+    }
     datadog_sigaction->sa_sigaction = handle_signal;
     // we will use the SA_ONSTACK mask here to handle the signal in a freshly new stack.
     datadog_sigaction->sa_flags = SA_SIGINFO | SA_ONSTACK;
 
     return true;
 }
 
-/*
- * Overriding the signal handlers from the new spawned thread. This will allow us to block (handle)
- * also the `SIQGUIT` signal triggered when the process is unresponsive - ANR.
- * This function should be called inside a locked mutex for Thread safety
- */
 bool override_native_signal_handlers() {
-    if (handlers_installed) {
```

Comment:
why we don't need the check for `handlers_installed` anymore? because we are not spawning new thread and there is no chance this method can be entered few times?

Reply:
Yes, this method is only called from `try_to_install_handlers` which is already guarded by a mutex and we are already doing the check there so there is no need here anymore. Before it was called from a different Thread. I am going to get some time later and rewrite this file in C++ with a proper class and clean it a bit more as it got a bit messy after all these changes.